wagtail/wagtailcore/management/commands/fixtree.py

```diff
@@ -24,12 +39,48 @@ def handle_noargs(self, **options):
 
         if bad_depth or bad_numchild:
             Page.fix_tree(destructive=False)
-            problems_found = True
+            any_problems_fixed = True
+
+        if orphans:
+            # The 'orphans' list as returned by treebeard only includes pages that are
+            # missing an immediate parent; descendants of orphans are not included.
+            # Deleting only the *actual* orphans is a bit silly (since it'll just create
+            # more orphans), so generate a queryset that contains descendants as well.
+            orphan_paths = Page.objects.filter(id__in=orphans).values_list('path', flat=True)
+            filter_conditions = []
+            for path in orphan_paths:
+                filter_conditions.append(Q(path__startswith=path))
+
+            # combine filter_conditions into a single ORed condition
+            final_filter = reduce(operator.or_, filter_conditions)
+
+            # build a queryset of all pages to be removed; this must be a vanilla Django
+            # queryset rather than a treebeard MP_NodeQuerySet, so that we bypass treebeard's
+            # custom delete() logic that would trip up on the very same corruption that we're
+            # trying to fix here.
+            pages_to_delete = models.query.QuerySet(Page).filter(final_filter)
+
+            self.stdout.write("Orphaned pages found:")
+            for page in pages_to_delete:
+                self.stdout.write("ID %d: %s" % (page.id, page.title))
+            self.stdout.write('')
+
+            if options.get('interactive', True):
+                yes_or_no = six.moves.input("Delete these pages? [y/N] ")
+                if yes_or_no.lower().startswith('y'):
+                    deletion_count = len(pages_to_delete)
+                    pages_to_delete.delete()
+                    print("%d orphaned page%s deleted." % (deletion_count, "s"[deletion_count==1:]))
```

Comment:
This should be changed to:
``self.stdout.write()``
or, since we're using Python 3's print statement:
``print("message", file=self.stdout)``